src/main/java/seedu/meal360/Ui.java

```diff
@@ -0,0 +1,141 @@
+package seedu.meal360;
+
+import java.time.format.DateTimeFormatter;
+
+public class Ui {
+
+    private static final int BOXWIDTH = 100;
+    private static final String SEPARATORCHAR = "-";
+
+    public void printSeparator() {
+        String separatorLine = SEPARATORCHAR.repeat(BOXWIDTH);
+        System.out.println(separatorLine);
+    }
+
+    public void printWelcomeMessage() {
+        String logo = " __  __          _ ____  __  __\n" + "|  \\/  |___ __ _| |__ / / / /  \\\n"
+                + "| |\\/| / -_) _` | ||_ \\/ _ \\ () |\n" + "|_|  |_\\___\\__,_|_|___/\\___/\\__/\n";
+        System.out.println("Welcome to Meal360, your ultimate Recipe Manager!\n" + logo);
+    }
+
+    public void printGoodbyeMessage() {
+        System.out.println(formatMessage("Bye. Hope to see you again soon!"));
+    }
+
+    public String formatMessage(String message) {
+        return String.format("| %-97s|", message);
+    }
+
+    public void printMessage(String message) {
+        String outputMessage = formatMessage(message);
+        System.out.println(outputMessage);
+    }
+
+    public void printRecipe(Recipe recipe) {
+        System.out.println(formatMessage("Name of recipe: " + recipe.getName()));
+        int index = 1;
+        for (String ingredient : recipe.getIngredients().keySet()) {
+            String outputMessage = String.format("%s(%d)", ingredient,
+                    recipe.getIngredients().get(ingredient));
+            System.out.println(formatMessage(index + ". " + outputMessage));
+            index++;
+        }
+    }
+
+    public void printWeeklyPlan(WeeklyPlan weeklyPlan) {
+        if (weeklyPlan.isEmpty()) {
+            printMessage("Your weekly plan is empty!");
+        } else {
+            printMessage("Here is your weekly plan:");
+            weeklyPlan.forEach((recipe, count) -> {
+                String outputMessage = String.format("%s x%d", recipe, count);
+                printMessage(outputMessage);
+            });
+        }
+
+    }
+
+    public void printWeeklyIngredients(WeeklyPlan weeklyPlan, RecipeList recipeList) {
+        if (weeklyPlan.isEmpty()) {
+            printMessage("Your weekly plan is empty!");
+        } else {
+            printMessage("Here are your weekly ingredients:");
+            weeklyPlan.forEach((recipe, count) -> {
+                Recipe currRecipe = recipeList.findByName(recipe.toString().trim());
+                for (String ingredient : currRecipe.getIngredients().keySet()) {
```

Comment:
Seems like a lot of actions happening here. Perhaps consider breaking down certain multiple functions into separate variables for easier readability (eg. String sampleName = urrRecipe.getIngredients().keySet())